### PR TITLE
Ghana

### PR DESCRIPTION
**Pull request by Vesihiisi**
Mapping: https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/gh_(en)
Preview: https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/gh_(en)/preview

Task: https://phabricator.wikimedia.org/T174441

### CODE REVIEW

File: importer/GhEn.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            lookup = [table[x]["items"]
+                      for x in table if x.lower() == spec_is_raw]
+            special_is_q = lookup[0][0]
+            if len(lookup) == 1:
+                self.substitute_statement("is", special_is_q)
+
+    def set_inception(self):
+        if self.has_non_empty_attribute("built"):
+            if utils.legit_year(self.built):
+                year_qualifier = {"time_value": {"year": self.built}}
```
Comment (lokal-profil) [inline importer/GhEn.py]: self.built is a string. Do we need it to be an int or is that ok?

File: importer/GhEn.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    def set_special_is(self):
+        if self.has_non_empty_attribute("original_function"):
+            spec_is_raw = self.original_function.lower()
+            table = self.data_files["is"]["mappings"]
+            lookup = [table[x]["items"]
+                      for x in table if x.lower() == spec_is_raw]
+            special_is_q = lookup[0][0]
+            if len(lookup) == 1:
+                self.substitute_statement("is", special_is_q)
```
Comment (lokal-profil) [inline importer/GhEn.py]: else add report

File: importer/GhEn.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+class GhEn(Monument):
+
+    def set_special_is(self):
+        if self.has_non_empty_attribute("original_function"):
+            spec_is_raw = self.original_function.lower()
+            table = self.data_files["is"]["mappings"]
+            lookup = [table[x]["items"]
+                      for x in table if x.lower() == spec_is_raw]
+            special_is_q = lookup[0][0]
+            if len(lookup) == 1:
```
Comment (lokal-profil) [inline importer/GhEn.py]: can there be multiple valid  P31's at once?

**Comment by lokal-profil** [comment]
With regards to P131 I missed that this thsi was done using `region`. Would it be safer to do it using `region-iso` together with a mapping based on P300?

File: importer/GhEn.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    def update_descriptions(self):
+        desc = "material cultural heritage site in Ghana"
+        self.add_description("en", desc)
+
+    def set_monuments_all_id(self):
+        """Map which column name in specific table to  ID in monuments_all."""
+        self.monuments_all_id = self.id
+
+    def exists_with_monument_article(self, language):
```
[the remark this answers is not in the document]
Comment (lokal-profil) [reply importer/GhEn.py]: Done

File: importer/GhEn.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        if region_q:
+            self.add_statement("located_adm", region_q)
+
+    def set_location(self):
+        location_q = utils.q_from_first_wikilink("en", self.location)
+        if location_q:
+            self.add_statement("location", location_q)
+
+    def set_heritage_id(self):
+        self.add_statement("wlm_id", self.id)
```
[the remark this answers is not in the document]
Comment (lokal-profil) [reply importer/GhEn.py]: Done

File: importer/GhEn.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    def set_adm_location(self):
+        region_q = utils.q_from_first_wikilink("en", self.region)
+        if region_q:
+            self.add_statement("located_adm", region_q)
+
+    def set_location(self):
+        location_q = utils.q_from_first_wikilink("en", self.location)
+        if location_q:
+            self.add_statement("location", location_q)
```
[the remark this answers is not in the document]
Comment (lokal-profil) [reply importer/GhEn.py]: Done

File: importer/GhEn.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        if self.has_non_empty_attribute("built"):
+            if utils.legit_year(self.built):
+                year_qualifier = {"time_value": {"year": self.built}}
+                self.add_statement(
+                    "inception", year_qualifier)
+
+    def set_adm_location(self):
+        region_q = utils.q_from_first_wikilink("en", self.region)
+        if region_q:
+            self.add_statement("located_adm", region_q)
```
[the remark this answers is not in the document]
Comment (lokal-profil) [reply importer/GhEn.py]: Done

File: importer/GhEn.py (hunk 2 above)
Comment (lokal-profil) [reply importer/GhEn.py]: done

File: importer/GhEn.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+            table = self.data_files["is"]["mappings"]
+            lookup = [table[x]["items"]
+                      for x in table if x.lower() == spec_is_raw]
+            special_is_q = lookup[0][0]
+            if len(lookup) == 1:
+                self.substitute_statement("is", special_is_q)
+
+    def set_inception(self):
+        if self.has_non_empty_attribute("built"):
+            if utils.legit_year(self.built):
```
[the remark this answers is not in the document]
Comment (lokal-profil) [reply importer/GhEn.py]: Done

File: importer/GhEn.py (hunk 1 above)
Comment (lokal-profil) [reply importer/GhEn.py]: Should be ok

File: importer/GhEn.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        if location_q:
+            self.add_statement("location", location_q)
+
+    def set_heritage_id(self):
+        self.add_statement("wlm_id", self.id)
+
+    def update_labels(self):
+        english = utils.remove_markup(self.name)
+        self.add_label("en", english)
+        if self.has_non_empty_attribute("alternative_names"):
```
[the remark this answers is not in the document]
Comment (lokal-profil) [reply importer/GhEn.py]: Spliting by `,` and adding them separately

File: importer/GhEn.py (hunk 3 above)
Comment (lokal-profil) [reply importer/GhEn.py]: Answer: No they are one-to-one maps

**Comment by Vesihiisi** [comment]
Updated preview up to here.

**Pull request merged**